lib/compare.js

```diff
@@ -60,6 +60,8 @@ const strictEqualArray = (arr1, arr2) => {
 };
 
 const strictEqualObject = (obj1, obj2) => {
+  if (obj1 === null && obj2 === null) return true;
```

Comment:
This can be simplified to
```js
if (obj1 === null || obj2 === null) {
  return obj1 === null && obj2 === null;
}
```
Or, you can fit this in one line, which is just a matter of personal preference:
```js
if (obj1 === null || obj2 === null) return obj1 === null && obj2 === null;
```
Also, should this check be added to `equalObject` function as well?

Reply:
Is it really simplification?

It seems `equalObject` also needs a change, thanks :+1:

Reply:
If you mean that simple cases: (1) both null, (2) one null and other not null, should immediately return respective true and false, then I propose following solution:
```
const null1 = obj1 === null;
const null2 = obj2 === null;
if (null1) return null1 === null2;
```

Reply:
Why do you want to change original code? Is it hard to understand, does it use more operations or maybe does it have more symbols? I don't totally understand the goal of that.
From my point of view writing edge cases explicitly is simpler to write, simpler to understand.

Reply:
Just compare
```
if (obj1 === null && obj2 === null) return true;
if (obj1 === null || obj2 === null) return false;
```
4 comparison operators and 2 logical operators and
```
const null1 = obj1 === null;
const null2 = obj2 === null;
if (null1) return null1 === null2;
```
3 or 4 comparison operators and no logical operators, it's shorter and clear

Reply:
@tshemsedinov, the implementation you provided in your comment won't work if `obj2` is `null`.

Reply:
Moreover:
```
const null1 = obj1 === null;
if (null1) {
  const null2 = obj2 === null;
  return null1 === null2;
}
```

Reply:
If we are trying to optimize and improve readability

Reply:
@belochub agree
```
const null1 = obj1 === null;
const null2 = obj2 === null;
if (null1 || null2) return null1 === null2;
```
3 or 4 comparison and 1 logical operator

Reply:
@tshemsedinov, both the [first](https://github.com/metarhia/metatests/pull/50#discussion_r191089062) and the [second](https://github.com/metarhia/metatests/pull/50#discussion_r191092660) implementations won't work correctly, because when `obj1` is not equal to `null`, this code will skip checking the `obj2` for `null` equality, and therefore the function will fail later like it did before this patch.
The implementation I proposed in https://github.com/metarhia/metatests/pull/50#discussion_r191087860 though, can also be improved by taking an approach similar to the one you followed in your proposal, take a look: 
```js
if (obj1 === null || obj2 === null) return obj1 === obj2;
```

@DzyubSpirit, as to why this is a simpler version, first of all, it's shorter, there is only one if statement, second of all,  in worst case scenario, there are 3 comparisons, as opposed to 4 in your implementation.
In my opinion, it is easier to understand, but since both of the implementations work exactly the same in the end, and it's just a matter of personal preference, I am not insisting on changing it.
In my first comment, I've just proposed a simplified alternative (which has also improved now, thanks to this discussion) and it's still up to you to use it or not.